src/client/method-unit-actions.c

```diff
@@ -308,6 +308,127 @@ static int method_thaw_unit_on(Client *client, char *node_name, char *unit) {
+        char *node_name = commands->opargv[0];
+        int units_count = 1;
+        int r = 0;
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *result = NULL;
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        while (commands->opargv[units_count] != NULL) {
+                printf("%i %s\n", units_count, commands->opargv[units_count]);
+                r = sd_bus_call_method(
+                                client->api_bus,
+                                BC_INTERFACE_BASE_NAME,
+                                client->object_path,
+                                NODE_INTERFACE,
+                                "ResetFailedUnit",
+                                &error,
+                                &result,
+                                "s",
+                                commands->opargv[units_count]);
+
+                if (r < 0) {
+                        fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                        return r;
+                }
+                units_count++;
+        }
+        return r;
+}
+
+static int method_reset_failed_on_node(UNUSED Client *client, UNUSED char *node_name) {
+
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *result = NULL;
+        int r = 0;
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        BC_INTERFACE_BASE_NAME,
+                        client->object_path,
+                        NODE_INTERFACE,
+                        "ResetFailed",
+                        &error,
+                        &result,
+                        "");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        return r;
+}
+
+static int method_reset_failed_on_all_nodes(UNUSED Client *client) {
+
+        int r = 0;
+        sd_bus *bus = NULL;
+
+        r = sd_bus_open_system(&bus);
+        if (r < 0) {
+                fprintf(stderr, "Failed to connect to system bus: %s", strerror(-r));
+                return r;
+        }
+
+        sd_bus_error error = SD_BUS_ERROR_NULL;
+        sd_bus_message *result = NULL;
+        r = sd_bus_call_method(
+                        bus,
+                        "org.eclipse.bluechi",
+                        "/org/eclipse/bluechi",
+                        "org.eclipse.bluechi.Controller",
+                        "ListNodes",
+                        &error,
+                        &result,
+                        "",
+                        NULL);
```

Comment:
Please use the predefined constants for the service/object/interface names:
```suggestion
        r = sd_bus_call_method(
                        bus,
                        BC_INTERFACE_BASE_NAME,
                        BC_OBJECT_PATH,
                        CONTROLLER_INTERFACE,
                        "ListNodes",
                        &error,
                        &result,
                        "",
                        NULL);
```